docs/source/manual/core.rst

```diff
@@ -453,6 +453,77 @@ exception was thrown).
+
+This can be done by running these dependency health checks periodically in the background on some schedule,
+and then aggregating the results of all of those checks into a single indicator of overall health. Certain
+dependencies may be critical to your application functioning, like a database that your service can't function without,
+but other dependencies may be more non-critical to your service being able to function (let's say a cache, that
+could be considered more of a nice to have than a necessity).
+
+Define the following health check configurations in your `config.yml` file:
+
+.. code-block:: yaml
+
+    health:
+      delayedShutdownHandlerEnabled: true
+      shutdownWaitPeriod: 10s
+      healthChecks:
+        - name: user-database
+          critical: true
+        - name: user-notifications-queue
+          critical: false
+          schedule:
+            checkInterval: 2500ms
+            downtimeInterval: 10s
+            failureAttempts: 2
+            successAttempts: 1
+        - name: user-cache
+          critical: false
+
+.. note::
+
+    This behavior was integrated from the `Dropwizard Health module`__. If you are migrating from that module
+    to the new Dropwizard core framework health code, you will want to refer to :ref:`upgrade-notes-dropwizard-2_1_x-health` the migration guide.
+
+
+.. _`Kubernetes readiness & liveness checks`: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+.. __`Dropwizard Health module`: https://github.com/dropwizard/dropwizard-health
+
+.. _man-core-health-status:
+
+Application Status
+------------------
+
+There are two types of status that are supported currently: Alive and Ready.
+
+* An ``alive`` status indicates the application is operating normally and does not need to be restarted to recover from
+  a stuck state. Long-running applications can eventually reach a broken state and cannot recover except by being
+  restarted (e.g. deadlocked threads).
+* A ``ready`` status indicates the application is ready to serve traffic. Applications can temporarily be unable to
+  serve traffic due to a variety of reasons, for example, an application might need to build/compute large caches
+  during startup or can critically depend on an external service.
+
+Long-running applications can eventually reach a broken state and cannot recover except by being restarted
+(e.g. deadlocked threads).
+
+An example of how you might query the health check, assuming you're using the default responder/responseProvider
+settings in configuration:
+https://<hostname>:<port>/health-check?type=<type>
+(replace `<type>` with `ready` or `alive`; defaults to `ready`)
+
+Querying for particular health checks
+https://<hostname>:<port>/health-check?type=<type>&name=<name>
+(replace ``<type>`` with ``ready`` or ``alive``; defaults to ``ready``)
+(replace ``<name>`` with the name of the health check to query. Multiple names can be provided)
```

Comment:
Are we trying to replace the functionality of the built-in /healthchecks endpoint on the AdminServlet?

Reply:
I'm leaving that as out of scope for this PR honestly. I think that's a bigger change.

I am hoping to provide similar information, maybe in a more opt-in fashion from this endpoint though.